src/FSharp.Core/event.fs

```diff
@@ -17,138 +17,180 @@ module private Atomic =
+
+    member x.Trigger(args: obj[]) =
+        match multicast with
         | null -> ()
         | d -> d.DynamicInvoke(args) |> ignore
-    member x.Publish = 
-        { new IDelegateEvent<'Delegate> with 
+
+    member x.Publish =
+        { new IDelegateEvent<'Delegate> with
             member x.AddHandler(d) =
                 Atomic.setWith (fun value -> System.Delegate.Combine(value, d)) &multicast
+
             member x.RemoveHandler(d) =
                 Atomic.setWith (fun value -> System.Delegate.Remove(value, d)) &multicast }
 
 type EventDelegee<'Args>(observer: System.IObserver<'Args>) =
-    static let makeTuple = 
+    static let makeTuple =
         if Microsoft.FSharp.Reflection.FSharpType.IsTuple(typeof<'Args>) then
             Microsoft.FSharp.Reflection.FSharpValue.PreComputeTupleConstructor(typeof<'Args>)
         else
-            fun _ -> assert false; null // should not be called, one-argument case don't use makeTuple function
+            fun _ ->
+                assert false
+                null // should not be called, one-argument case don't use makeTuple function
 
-    member x.Invoke(_sender:obj, args: 'Args) = 
-        observer.OnNext args
-    member x.Invoke(_sender:obj, a, b) = 
-        let args = makeTuple([|a; b|]) :?> 'Args
+    member x.Invoke(_sender: obj, args: 'Args) = observer.OnNext args
+
+    member x.Invoke(_sender: obj, a, b) =
+        let args = makeTuple ([| a; b |]) :?> 'Args
         observer.OnNext args
-    member x.Invoke(_sender:obj, a, b, c) = 
-        let args = makeTuple([|a; b; c|]) :?> 'Args
+
+    member x.Invoke(_sender: obj, a, b, c) =
+        let args = makeTuple ([| a; b; c |]) :?> 'Args
         observer.OnNext args
-    member x.Invoke(_sender:obj, a, b, c, d) = 
-        let args = makeTuple([|a; b; c; d|]) :?> 'Args
+
+    member x.Invoke(_sender: obj, a, b, c, d) =
+        let args = makeTuple ([| a; b; c; d |]) :?> 'Args
         observer.OnNext args
-    member x.Invoke(_sender:obj, a, b, c, d, e) = 
-        let args = makeTuple([|a; b; c; d; e|]) :?> 'Args
+
+    member x.Invoke(_sender: obj, a, b, c, d, e) =
+        let args = makeTuple ([| a; b; c; d; e |]) :?> 'Args
         observer.OnNext args
-    member x.Invoke(_sender:obj, a, b, c, d, e, f) = 
-        let args = makeTuple([|a; b; c; d; e; f|]) :?> 'Args
+
+    member x.Invoke(_sender: obj, a, b, c, d, e, f) =
+        let args = makeTuple ([| a; b; c; d; e; f |]) :?> 'Args
         observer.OnNext args
 
 
-type EventWrapper<'Delegate,'Args> = delegate of 'Delegate * obj * 'Args -> unit
+type EventWrapper<'Delegate, 'Args> = delegate of 'Delegate * obj * 'Args -> unit
 
 [<CompiledName("FSharpEvent`2")>]
-type Event<'Delegate, 'Args when 'Delegate : delegate<'Args, unit> and 'Delegate :> System.Delegate and 'Delegate: not struct>() =  
+type Event<'Delegate, 'Args when 'Delegate: delegate<'Args, unit> and 'Delegate :> System.Delegate and 'Delegate: not struct>
+    () =
```

Comment:
Is it expected to have constructor moved to new line?

Reply:
I guess so.  I mean it's at column 133 and it needs a break :)

@nojaf will the adjusted constraint formatting change this? e.g.

```
type Event<'Delegate, 'Args
                        when 'Delegate: delegate<'Args, unit>
                        and 'Delegate :> System.Delegate
                        and 'Delegate: not struct> () =
```

Reply:
Yeah, I guess that's what I meant, is if it's possible, to have line breaks at the `and`s.

Reply:
No, probably not. I created https://github.com/fsprojects/fantomas/issues/2266.